integration_tests/suite/test_phonebook_http.py

```diff
@@ -358,3 +391,304 @@ def test_get_multi_tenant(self, sub, main):
             self.assert_unknown_source_exception(main['uuid'], e)
         else:
             self.fail('Should have raised')
+
+
+class TestGetContacts(BasePhonebookCRUDTestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        super().setUpClass()
+        cls.phonebook_crud = PhonebookCRUD(cls.Session)
+        cls.contact_crud = PhonebookContactCRUD(cls.Session)
+        cls.stack = ExitStack()
+
+        cls.phonebook = cls.phonebook_crud.create(
+            MAIN_TENANT,
+            {'name': 'test-phonebook', 'description': 'some test phonebook'},
+        )
+        cls.stack.callback(
+            cls.phonebook_crud.delete, None, PhonebookKey(uuid=cls.phonebook['uuid'])
+        )
+        cls.contacts, errors = cls.contact_crud.create_many(
+            [MAIN_TENANT],
+            PhonebookKey(uuid=cls.phonebook['uuid']),
+            [
+                {
+                    'firstname': f'Contact {i}',
+                    'lastname': 'McContact',
+                    'number': str(1000000000 + i),
+                }
+                for i in range(5000)
```

Comment:
Open to discussion: having a lot of payload can make debugging the tests much harder. I would prefer separating logic tests (with smaller payloads) from load tests (with larger payloads) that only do timing assertions.